### PR TITLE
feat: add desc share sql

### PR DESCRIPTION
**Pull request by lichuang**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

feat: add desc share sql

Fixes https://github.com/datafuselabs/databend/issues/7104

### CODE REVIEW

**Comment by Xuanwo** [comment]
What's the meaning of `desc share a.b`? We can desc shares of other tenants?

File: query/tests/it/interpreters/interpreter_share_desc.rs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    // show create share
+    {
+        let query = "desc share t";
+        let (plan, _, _) = planner.plan_sql(query).await?;
+        let executor = InterpreterFactoryV2::get(ctx.clone(), &plan)?;
+        assert_eq!(executor.name(), "DescShareInterpreter");
+        let stream = executor.execute().await?;
+        let result = stream.try_collect::<Vec<_>>().await?;
+        let expected = vec![
+            "+------+------+-----------+",
```
Comment (BohuTANG) [inline query/tests/it/interpreters/interpreter_share_desc.rs:45]: Why the results empty?

**Comment by lichuang** [comment]
> What's the meaning of `desc share a.b`? We can desc shares of other tenants?

in https://databend.rs/doc/contributing/rfcs/share#describe-share there is `desc share` rfc:

```
Describes the shared objects that are included in a share.
```

File: query/tests/it/interpreters/interpreter_share_desc.rs (hunk 1 above)
Comment (lichuang) [reply query/tests/it/interpreters/interpreter_share_desc.rs:45]: > Why the results empty?

in https://databend.rs/doc/contributing/rfcs/share#describe-share there is `desc share` rfc:

```
Describes the shared objects that are included in a share.
```

cause column `grant_on` cannot be determined each time the test run, so just test an empty share without any shared objects.

File: common/ast/src/ast/statements/share.rs [hunk 2]
```diff
@@ -127,3 +127,18 @@ impl Display for AlterShareTenantsStmt<'_> {
...
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct DescShareStmt<'a> {
+    pub share: Identifier<'a>,
+}
+
+impl Display for DescShareStmt<'_> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "DESC SHARE ")?;
+
+        write!(f, "{}", self.share)?;
```
Comment (Xuanwo) [inline common/ast/src/ast/statements/share.rs:140]:
```suggestion
        write!(f, "DESC SHARE {}", self.share)?;
```

File: common/ast/src/parser/statement.rs [hunk 3]
```diff
@@ -901,6 +906,7 @@ pub fn statement(i: Input) -> IResult<StatementMsg> {
             | #grant_share_object: "`GRANT { USAGE | SELECT | REFERENCE_USAGE } ON { DATABASE db | TABLE db.table } TO SHARE <share_name>`"
             | #revoke_share_object: "`REVOKE { USAGE | SELECT | REFERENCE_USAGE } ON { DATABASE db | TABLE db.table } FROM SHARE <share_name>`"
             | #alter_share_tenants: "`ALTER SHARE [IF EXISTS] <share_name> { ADD | REMOVE } TENANTS = tenant [, tenant, ...]`"
+            | #desc_share: "`{DESC | DESCRIBE} SHARE [tenant.]<share_name>`"
```
Comment (Xuanwo) [inline common/ast/src/parser/statement.rs:909]:
```suggestion
            | #desc_share: "`{DESC | DESCRIBE} SHARE <share_name>`"
```

File: common/ast/tests/it/parser.rs [hunk 4]
```diff
@@ -268,6 +268,7 @@ fn test_statement() {
         r#"ALTER SHARE a ADD TENANTS = b,c;"#,
         r#"ALTER SHARE IF EXISTS a ADD TENANTS = b,c;"#,
         r#"ALTER SHARE IF EXISTS a REMOVE TENANTS = b,c;"#,
+        r#"DESC SHARE b;"#,
```
Comment (Xuanwo) [inline common/ast/tests/it/parser.rs:271]: Add a test case for `DESCRIBE`

**Comment by lichuang** [comment]
@Xuanwo

**Review by Xuanwo — APPROVED** [review]

**Review by BohuTANG — APPROVED** [review]

**Pull request merged**